homeworks/week2/Hong-Nguyen.java

```diff
@@ -0,0 +1,30 @@
+package com.Hiento.session2;
+
+import java.util.HashSet;
+import java.util.Set;
+
+// Time complexity is O(n) because you need to go through each and every letter.
+// Space complexity is O(1) because the number of letters is fixed
+
+public class RemoveDuplicate {
+    public static void main(String[] args) {
+        char[] s = {'h', 'e', 'l', 'l', 'o', ' ', 'w', 'o', 'r', 'l', 'd'};
+        removeDuplicateCharacters(s);
+        // s = {'h', 'e', 'l', 'o', ' ', 'w', 'r', 'l', 'd'};
+        for (int i = 0; i < s.length; i++) {
+            System.out.print(s[i]);
+        }
+        // Output: "helo wrd"
+    }
+
+    public static void removeDuplicateCharacters(char[] s) {
+        Set<Character> seen = new HashSet<>();
+        for (int i = 0; i < s.length; i++) {
+            if (seen.contains(s[i])) {
+                s[i] = 0;
```

Comment:
It's always a plus when candidate add tests without being asked. In this context, it also tells us that you have tested your code.

Reply:
> Are you trying to remove this character from the input string? If yes, I don't think this will work.
> You can try to run some tests to see if this method returns correctly.

I have tested and it worked.

Reply:
> It's always a plus when candidate add tests without being asked. In this context, it also tells us that you have tested your code.

thank you, i'm doing now

Reply:
I have edited the code in githut, and click proposed changed. However, I dont see the code is changed , not sure you need to click Accept changes or something? Or should I push code again?

Reply:
I tried to push again and have error 
![image](https://user-images.githubusercontent.com/67253556/85937681-dbd7c400-b8ba-11ea-9db8-4195d2d567cf.png)

Reply:
I'm not sure about this error. Let's merge this code now and you can submit your change later.